### DISCUSSION TITLE
[Improvement Idea] Overhaul Retro Ratio "White Points"

### DISCUSSION THREAD

**Discussion by Xeinok-Pals** [RetroAchievements/RAWeb#1209] [Ideas]
EDIT: 2024-01-18 Please note that some of the items mentioned below have been updated over the years. I no longer think that "ratio" as a whole should be removed since the new way to view it: "x5.37 Rarity" can be a nice quick way to view games in this context.

Hello, I would like to have some discussion around the issues I see with the Retro Ratio points system, and some ideas for improvements. I do think that this "true" rarity-based point system is very important and that they should be supported and featured a bit more. These are just my personal opinions and ideas, so all feedback is welcome! I will refer to Retro Ratio Points below as "White Points" as well.

Other references:
https://docs.retroachievements.org/FAQ/#what-are-the-white-points (How docs currently describes White Points)
https://retroachievements.org/viewtopic.php?t=1015 (2014 thread on White Points)
https://github.com/RetroAchievements/RAWeb/pull/504 (Xymjak proposed fixes)

**### Main Issue 1: General Confusion**
I've seen many users on discord express confusion with questions like "what is the number after my points number?", etc. A huge part of the confusion is simply in the name "Retro Ratio".

**Solutions:**
1. Hover-over text should appear when you hover over Hardcore Points and White Points which explains what each are. This should be done for game pages as well as profile pages and top-right profile: 
![image](https://user-images.githubusercontent.com/65511279/198131957-539634bc-5b99-420d-8154-0d08233ff341.png)
2. I propose that "Retro Ratio" should be renamed to something such as "Challenge Points" or "Rarity Points" which more intuitively explains their purpose and removes focus from the "ratio" part.

**### Main Issue 2: Lack of Support for White Points**
Unlike Hardcore and Softcore Point Ranks, you cannot click easily into any sort of leaderboard view for White Points.
![image](https://user-images.githubusercontent.com/65511279/198129752-09bfcf6a-8f83-46d0-b556-5ea72c1fbb0e.png)

**Solutions:**
1. White Points should be broken out to their own section under Hardcore Points, showing their own clickable Rank and percentage placement.
2. White Points should be displayed on recently played game lists on profile: 
![image](https://user-images.githubusercontent.com/65511279/198136426-4004f332-cccd-46ac-a7ce-8740141750ab.png)
3. Game Lists should be sortable by White Points: 
![image](https://user-images.githubusercontent.com/65511279/198136765-1d30859d-2ad7-41ae-8afa-368b02b319f9.png)
4. Game pages should allow achievements to be sortable by White Points: 
![image](https://user-images.githubusercontent.com/65511279/198136881-d324a3fa-4006-4836-a19f-71fbaf906470.png)

**### Main Issue 3: Problems with Focus on "Ratio"**
In an accumulative hobby such as achievement hunting, where a person amasses more and more achievements/masteries, I'm not sure that a "ratio" of "easy achievements vs. hard achievements" (in actuality, it is common vs. rare, of course) is a worthwhile metric to focus on so prominently. In many ways, this incentivizes unwanted behaviors such as players locking "easy" achievements after earning harder ones in a game to boost RR, avoiding less popular games such as romhacks, etc. where the small playerbase will typically master the game and thus reduce the RR, etc.
![image](https://user-images.githubusercontent.com/65511279/198137452-89371ccd-fa54-4c17-adb2-e7a319ceacf3.png)
![image](https://user-images.githubusercontent.com/65511279/198151618-27db2210-f4cb-404b-9caa-dd5923895f98.png)

A "ratio" metric can be very confusing in situations such as this:
Player 1 has mastered 500 games - 100 of them were brutally difficult and worth many white points.
Player 2 has mastered only 20 games, but all 20 of them were worth many white points.

In this situation, Player 2 will appear much better in terms of "ratio" when the truth is that Player 1 has completed 5x as many "difficult" games on top of many other more common games. I think that focusing on "ratio" in an accumulative hobby such as this can just lead to more confusion, potential for elitism about types of games played, etc.

**Solutions:**
1. Remove Retro Ratio from player pages and instead focus on a more total, accumulative White Points Rank as discussed in "Main Issue 2" above.

Some other ideas may be added in the future - thanks!

**Comment by Jamiras** [comment]
Some other ideas for improving retro ratio are in #664, and there was even an attempt to implement some of the changes made in #504.

The biggest problem is that it's completely dependent on the number of players that a set has. As you've pointed out, some sets only have a few players that unlocked most of the achievements, even if they're really hard. On the other end of the spectrum, you have some sets that have been played by thousands of people who moved on after playing for thirty minutes. Their handful of unlocks artificially inflates the rest of the set. And the worst issue is revisions. When new achievements are added to a set, the hundreds (or thousands) of players who had previously played the set suddenly don't have the new achievements, so they seem really hard/rare.

**Reply by Xeinok-Pals** [reply]
Thanks for referring to the related discussions!

I focused more on web presence/overall support in this post (because I DO very much enjoy White Points and think that they are an important part of an achievement site), but I do have some thoughts more in line with #664 and #504 around the specific implementation of a rarity-based points system and its specific formula. I will give those threads a good read and add comments.

There's a lot of great resources out there to pull examples from on this kind of working formula such as the TrueAchievements/TrueNetwork system and also the Steam Hunters system (my personal favorite at this time). Here is some out-of-date material around the SH rarity-based points system: https://www.patreon.com/posts/heres-how-steam-32150494 & https://www.patreon.com/posts/upcoming-update-45005109. Rudey, the dev of SH, might be open to some conversations as well.

**Reply by Xeinok-Pals** [reply]
Specifically regarding games getting White Points "inflated" by revisions - I am personally 100% okay with this.

1. It almost always quickly deflates back down over the following 1-2 months
2. It serves as an incentive to promote a wanted behavior: players re-mastering their games
3. If White Points were renamed as "Rarity Points" and clearly explained in a way which makes it clear that it does not perfectly represent "difficulty" and how factors such as obscurity/grind/difficulty/popularity of game/etc. ALL play a part in the valuation, then the userbase would likely be more okay with it

**Comment by Xeinok-Pals** [comment]
Could someone write out the specific logic/formula for White Points as they currently exist on RA please?

**Reply by Tsearo** [reply]
https://docs.retroachievements.org/FAQ/#what-are-the-white-points
This forum details the formula: https://retroachievements.org/viewtopic.php?t=1015
p = points
a = achievers
t = total users who have played the game
r = 0.6
RP = p * r + ( p * ( t / a ) * ( 1 - r ) )

**Reply by Jamiras** [reply]
The original discussion is [here](https://retroachievements.org/viewtopic.php?t=1015).

I think that matches the code (note `$ratioFactor` is `0.4` vs. the `0.6` in the forum post, so effectively `r` and `1-r` are swapped):
https://github.com/RetroAchievements/RAWeb/blob/2f7e972732cf7317c04693fa20e6a0654397f1b4/lib/database/achievement-points.php#L36-L37

**Reply by Xeinok-Pals** [reply]
and just confirming that for now, these proposed changes are still not in prod? https://github.com/RetroAchievements/RAWeb/pull/504

**Reply by Jamiras** [reply]
Correct. There were several changes requested that were never implemented, and now the PR is quite out-of-date, and has conflicts that need to be resolved.

**Comment by Xeinok-Pals** [comment]
It might be nice if we could have Xymjak come and re-explain his ideas in the 2024 context and it would also be wonderful if we could get Rudey, the creator of https://steamhunters.com/ to come and speak on this subject as SH has a quite elegant formula for "true rarity points" as discussed above.
